packages/core/saved-objects/core-saved-objects-migration-server-internal/src/zdt/next.ts

```diff
@@ -59,19 +79,108 @@ export const nextActionMap = (context: MigratorContext) => {
         timeout: '60s',
       }),
     UPDATE_MAPPING_MODEL_VERSIONS: (state: UpdateMappingModelVersionState) =>
-      Actions.updateMappings({
+      Actions.updateIndexMeta({
         client,
         index: state.currentIndex,
-        mappings: {
-          properties: {},
-          _meta: state.currentIndexMeta,
-        },
+        meta: setMetaMappingMigrationComplete({
+          meta: state.currentIndexMeta,
+          versions: context.typeModelVersions,
+        }),
       }),
     UPDATE_ALIASES: (state: UpdateAliasesState) =>
       Actions.updateAliases({
         client,
         aliasActions: state.aliasActions,
       }),
+    INDEX_STATE_UPDATE_DONE: (state: IndexStateUpdateDoneState) => () => Actions.noop(),
+    DOCUMENTS_UPDATE_INIT: (state: DocumentsUpdateInitState) => () => Actions.noop(),
+    SET_DOC_MIGRATION_STARTED: (state: SetDocMigrationStartedState) =>
+      Actions.updateIndexMeta({
+        client,
+        index: state.currentIndex,
+        meta: setMetaDocMigrationStarted({
+          meta: state.currentIndexMeta,
+        }),
```

Comment:
This is sometimes awkward in the v2 migration model too :( it's not great that the previous stage needs to know how to prepare the state for the next stage. Almost feels like we want to have two callbacks for each stage in the model, one when entering that state and another when we get an action response. But even if we try this, it feels like the kind of thing we don't want to do inside another PR anyway.

The downside of adding it here is that there's business logic outside of the model and outside the actions making it harder to understand and test. I think we should either
1. add more test coverage to next.ts
1. Move this into a dedicated `Actions.setDocMigrationStarted` and test it there

I'd slightly lean towards (2) but this does also mean we're putting more business logic into actions instead of them 🤷

Reply:
Yeah, I'm still unsure what the best way of doing this would be. Specialized actions make sense, but I'm afraid specializing too much may not be the correct way either...

I'm going to just add test coverage around `nextActionMap` for now